iris-mpc/src/bin/server.rs

```diff
@@ -1095,6 +1102,11 @@ async fn server_main(config: Config) -> eyre::Result<()> {
         let dummy_shares_for_deletions = get_dummy_shares_for_deletion(party_id);
 
         loop {
+            if shutdown_handler.is_shutting_down() {
+                tracing::info!("Exiting from main loop gracefully due to shutdown signal...");
+                return Ok(());
+            }
+
```

Comment:
Won't this now leave one batch unprocessed like this, since we start fetching the next batch already in the previous loop iteration?

Reply:
Should we maybe rather have this `is_shutting_down()` check around the `receive_batch` call (which fills the next batch), set the `next_batch` None, the the current batch process and exit the loop in the next iteration if `next_batch` is None?